mountpoint-s3-client/src/endpoint_config.rs

```diff
@@ -1,9 +1,13 @@
+use std::os::unix::prelude::OsStrExt;
+
 use mountpoint_s3_crt::{
     common::{allocator::Allocator, uri::Uri},
-    s3::endpoint_resolver::{RequestContext, ResolverError, RuleEngine},
+    s3::endpoint_resolver::{RequestContext, ResolvedEndpoint, ResolverError, RuleEngine},
 };
 use thiserror::Error;
 
+use crate::s3_crt_client::AuthScheme;
```

Comment:
I think `AuthScheme` probably belongs in this file. You can make the fields `pub(crate)` if you need to read them from `s3_crt_client`.